portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java

```diff
@@ -23,8 +40,29 @@ public void init() {
     // Gets the log in status.
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("text/html;");
+        response.setContentType("application/json;");
+
+        boolean isLoggedIn = this.userService.isUserLoggedIn();
+        String loggingUrl;
+        String user;
+
+        if (isLoggedIn) {
+            user = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = URL_TO_REDIRECT_TO_AFTER_LOGS_OUT;
+            loggingUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+        } else {
+            user = GUEST_USER_NAME;
+            String urlToRedirectToAfterUserLogsIn = URL_TO_REDIRECT_TO_AFTER_LOGS_IN;
+            loggingUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+        }
+
+        LoginResult loginResult = new LoginResult(isLoggedIn, loggingUrl, user);
+        response.getWriter().println(convertToJsonUsingGson(loginResult));
+    }
 
-        response.getWriter().println(this.userService.isUserLoggedIn());
+    private String convertToJsonUsingGson(LoginResult loginResult) {
```

Comment:
I thought you created a util method for this?

Reply:
i created in the next PR haha